src/chat/chat_history_card.rs

```diff
@@ -360,8 +360,14 @@ impl Widget for ChatHistoryCard {
             .to_uppercase()
             .to_string();
 
-        match chat.borrow().associated_entity {
-            Some(ChatEntityId::Agent(agent)) => {
+        match &chat.borrow().associated_entity {
+            Some(ChatEntityId::Agent(agent_id)) => {
+                let agent = store
+                    .chats.available_agents
+                    .get(&agent_id)
+                    .cloned()
+                    .unwrap_or_default();
```

Comment:
I would divide this flow to handle the case of agent missing, or at least, if you really want to do this, I would make a constructor `MofaAgent::unknown()` or something similar instead of using the default as that

choose your pill

Reply:
So this case is not about starting new chats with an inaccesible agent but rather about how we handle the chat history whenever we're referring to an agent that is no longer reachable.

I agree that this was not very clean or clear, I've updated (across all the files) to:
```rs
let agent = store.chats.get_agent_or_placeholder(&agent_id);
```
With:
```rs
/// Returns a reference to an agent by ID, falling back to an unknown agent placeholder
/// if the agent is not found in the available agents list.
/// 
/// This is useful when dealing with historical chat references to agents that may
/// no longer be available (e.g., server disconnected or agent deleted).
/// 
/// In the future, we'll want a more sophisticated solution, by potentially storing 
/// agent information locally and updating it when a server is connected.
pub fn get_agent_or_placeholder(&self, agent_id: &AgentId) -> &MofaAgent {
    self.available_agents.get(agent_id).unwrap_or(&self.unknown_agent)
}
```

As mentioned there this is a temporary solution while we wait for confirmation from the MoFa team on what information they will provide, how could we keep a local copy of some data, sync, etc. Right now you cannot start a new chat with an unreachable agent which is the most important part.
Besides that, we should improve the error handling (mostly notify the user properly) of errors in either WasmEdge backend models or MoFa. Currently for either of them you just don't see anything happening on the screen, but we should fix that for both cases, perhaps in a separate PR.